### PR TITLE
VsTestTask Changes for enabling XPlat Code Coverage(via dotnet test)

### PR DESCRIPTION
**Pull request by vagisha-nidhi**
## Description
Adding **VSTestXPlatDataCollectorDirectoryPath** to be added to test adapter path for picking XPlat code coverage data collector in case of `--collect:"xplat code coverage"`.
`XPlatDataCollectorDirectoryPath `variable is set by the data collector in its `props `file.

### CODE REVIEW

File: src/Microsoft.TestPlatform.Build/Resources/Resources.resx [hunk 1]
```diff
@@ -129,6 +129,9 @@
   <data name="TestRunningSummary" xml:space="preserve">
     <value>Test run for {0}({1})</value>
   </data>
+  <data name="UpdateTemplateForCollectingXPlatCodeCoverage" xml:space="preserve">
+    <value>Warning: Update the test project template and add reference to Microsoft.TestPlatform.Extensions.CoverletCoverageDataCollector to collect xplat code coverage</value>
```
Comment (singhsarab) [inline src/Microsoft.TestPlatform.Build/Resources/Resources.resx]: What do we mean by updating the Test project template ?

File: src/Microsoft.TestPlatform.Build/Tasks/VSTestTask.cs [hunk 2]
```diff
@@ -268,47 +275,57 @@ private List<string> AddArgs()
...
+
+        /// <summary>
+        /// Adds code coverage path in test adapter.
+        /// </summary>
+        /// <param name="isCollectCodeCoverageEnabled"></param>
+        /// <param name="isRunSettingsEnabled"></param>
+        /// <param name="adapterPath"></param>
+        /// <param name="allArgs"></param>
+        /// <param name="warningMessage"></param>
+        private void AddCodeCoveragePathInTestAdapter(bool isCollectCodeCoverageEnabled, bool isRunSettingsEnabled, string adapterPath, List<string> allArgs, string warningMessage)
```
Comment (mayankbansal018) [inline src/Microsoft.TestPlatform.Build/Tasks/VSTestTask.cs]: The job is this method should simply to add a given adapter path to arguments. It doesn't need to check for runsettings, & coverage. Make these checks before invoking

File: src/Microsoft.TestPlatform.Build/Tasks/VSTestTask.cs [hunk 3]
```diff
@@ -268,47 +275,57 @@ private List<string> AddArgs()
             {
                 foreach (var arg in this.VSTestCollect)
                 {
-                    if (arg.Equals("Code Coverage", StringComparison.OrdinalIgnoreCase))
-                    {
-                        isCollectCodeCoverageEnabled = true;
-                    }
+                    isCollectCodeCoverageEnabled = arg.Equals("Code Coverage", StringComparison.OrdinalIgnoreCase);
```
Comment (mayankbansal018) [inline src/Microsoft.TestPlatform.Build/Tasks/VSTestTask.cs]: >isCollectCodeCoverageEnabled [](start = 20, length = 28)

These variables can be done away with, just invoke below methods, & the string matches

**Review by mayankbansal018 — CHANGES_REQUESTED** [review]
:clock1:

File: src/Microsoft.TestPlatform.Build/Resources/Resources.resx (hunk 1 above)
Comment (vagisha-nidhi) [reply src/Microsoft.TestPlatform.Build/Resources/Resources.resx]: We will be adding the datacollector in csproj like `<PackageReference Include="Microsoft.TestPlatform.Extensions.CoverletCoverageDataCollector" Version="1.0.0" />`
This meant adding the reference.

File: src/Microsoft.TestPlatform.Build/Tasks/VSTestTask.cs (hunk 2 above)
Comment (vagisha-nidhi) [reply src/Microsoft.TestPlatform.Build/Tasks/VSTestTask.cs]: Done. Would need to send if coverage is enabled or not for showing the warning message specific to `code coverage` or `xplat code coverage`

File: src/Microsoft.TestPlatform.Build/Tasks/VSTestTask.cs (hunk 3 above)
Comment (vagisha-nidhi) [reply src/Microsoft.TestPlatform.Build/Tasks/VSTestTask.cs]: Removed the variables

File: src/Microsoft.TestPlatform.Build/Tasks/VSTestTask.cs [hunk 4]
```diff
@@ -115,6 +115,12 @@ public string VSTestTraceDataCollectorDirectoryPath
             set;
         }
 
+        public string VSTestXPlatDataCollectorDirectoryPath
```
Comment (mayankbansal018) [inline src/Microsoft.TestPlatform.Build/Tasks/VSTestTask.cs]: >VSTestXPlatDataCollectorDirectoryPath [](start = 22, length = 37)

Have we thought of unifying the arguments, & convert it to List<string>?

File: src/Microsoft.TestPlatform.Build/Resources/Resources.resx (hunk 1 above)
Comment (singhsarab) [reply src/Microsoft.TestPlatform.Build/Resources/Resources.resx]: it seems redundant then,
Change it to "To collect xplat code coverage please add reference to Microsoft.TestPlatform.Extensions.CoverletCoverageDataCollector to the test project."

And we can add more information and a link we plan to blog it.

**Comment by vagisha-nidhi** [comment]
We will be appending the datacollector path as part of VSTestTestAdapterPath in datacollector targets file. So no need of any change here.

**Pull request closed without merging**